src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -117,6 +120,17 @@ internal void Update(long number)
                         break;
                     }
             }
+
+            // There is a race with SnapShot:
+            // Update updates value
+            // SnapShot snapshots the value
+            // SnapShot sets status to NoCollectPending
+            // Update sets status to CollectPending -- this is not right as the SnapShot
+            // already included the updated value.
+            // In the absence of any new Update call until next SnapShot,
+            // this results in exporting an Update even though
+            // it had no update.
```

Comment:
For delta we can probably mitigate it by filtering out the zero points before feeding it to the MetricReader. Doesn't have to be solved right now (maybe leave a comment?).